b1ddi/resource_ipam_address_test.go

```diff
@@ -71,6 +71,60 @@ func resourceAddressBasicTestStep() resource.TestStep {
 	}
 }
 
+func resourceSubnetNAIPBasicTestStep() resource.TestStep {
+	return resource.TestStep{
+		Config: `
+					resource "b1ddi_ip_space" "tf_acc_test_space" {
+  						name = "tf_acc_test_space"
+  						comment = "This IP Space is created by terraform provider acceptance test"
+					}
+					resource "b1ddi_address_block" "tf_address_block"{
+						name = "tf_address_block"
+						address = "192.168.0.0"
+   						cidr = 24
+						space = b1ddi_ip_space.tf_acc_test_space.id
+						comment = "This address block is created by terraform provider acceptance test"
+					}
+					resource "b1ddi_subnet" "tf_acc_test_subnet" {
+						name = "tf_acc_test_subnet"						
+						address = "192.168.1.0"
+						space = b1ddi_ip_space.tf_acc_test_space.id
+						cidr = 24
+  						comment = "This Subnet is created by terraform provider acceptance test"
+					}
+					resource "b1ddi_range" "tf_acc_test_range" {
+						start = "192.168.1.15"
+						end = "192.168.1.30"
+  						name = "tf_acc_test_range"
+						space = b1ddi_ip_space.tf_acc_test_space.id 
+  						comment = "This Range is created by terraform provider acceptance test"
+						depends_on = [b1ddi_subnet.tf_acc_test_subnet]
```

Comment:
Why having `depends_on` only defined for this object?

Reply:
Becuase this resource requires the subnet `tf_acc_test_subnet` to be created. While this may not be required, it ensures the address is created after the subnet object is created.

Reply:
Got that yes but to create the subnet you also need the IP Space being created before and I don't see `depends_on` defined there. Or Terraform figures this itself because the IP Space is explicitly mentioned in the subnet creation request?

Reply:
Yes, I believe so. Never had issues when an object had only a space reference. Since Range is dependent on the Subnet block, the creation ordering isn't followed everytime. `depends_on` added to ensure this block is excuted after Subnet always.